assets/app/scripts/directives/podStatusChart.js

```diff
@@ -109,18 +119,11 @@ angular.module('openshiftConsole')
             }
           });
 
-          angular.forEach(phases, function(phase) {
-            columns.push([phase, countByPhase[phase] || 0]);
-          });
-
-          chart.load({ columns: columns });
+          return countByPhase;
         }
 
-        $timeout(function() {
-          chart = c3.generate(config);
-          $scope.$watch('pods', updateChart, true);
-          $scope.$watch('desired', updateCenterText);
-        }, 0, false);
+        $scope.$watch(countPodPhases, updateChart, true);
```

Comment:
By default, watch will compare all of the fields of all of the pods to see if anything has changed. The custom watch function should be more efficient because it only looks at pod phase and ignores other fields.